contracts/CompoundMiddleContract.sol

```diff
@@ -191,121 +119,177 @@ contract CompoundMiddleContract {
         require(error2 == 0, "comptroller.getAccountLiquidity FAILED");
         require(shortfall == 0, "account underwater");
         require(liquidity > 0, "account has excess collateral");
 
         console.log("Liquidity available", liquidity); // liquidity is the USD value borrowable by the user, before it reaches liquidation
 
         // borrow
-        uint256 amountToBorrowInWei = 2000000000000000; // CHANGE
-        cEth.borrow(amountToBorrowInWei);
+        require(cEth.borrow(_amountToBorrowInWei) == 0, "BORROW FAILED");
         uint256 borrowBalance = cEth.borrowBalanceCurrent(address(this));
 
-        console.log("Borrowed %s eth", amountToBorrowInWei / 18);
+        console.log("Borrowed %s eth", _amountToBorrowInWei / 18);
+
+        (bool success, ) = owner.call{ value: address(this).balance }("");
+
+        require(success, "FAILURE, ETHER NOT SENT");
 
         return borrowBalance;
     }
 
     /**
      * @dev repays borrowed eth
      * @param _cEtherAddress address of the cEth contract in Compound
-     * @param amount amount of ether to be repayed
-     * @return bool status of transaction
+     * @return uint256 updated borrow balance of the user
      */
-    function paybackEth(address _cEtherAddress, uint256 amount)
-        public
-        returns (bool)
-    {
+    function paybackEth(
+        address _cEtherAddress,
+        uint256 gas
+    ) public payable returns (uint256) {
         CEth cEth = CEth(_cEtherAddress);
-        cEth.repayBorrow{value: amount}();
+        cEth.repayBorrow{value: msg.value, gas: gas}();
 
-        console.log("Repayed %s ether", amount);
+        console.log("Repayed %s ether", msg.value);
 
-        return true;
+        return cEth.borrowBalanceCurrent(address(this));
     }
 
     /**
-     * @dev deposits erc20 tokens to Compound, mints cTokens 
+     * @dev deposits erc20 tokens to Compound, mints cTokens
      * @param _erc20Contract address of the erc20 token contract
-     * @param _cErc20Contract address of the Compound contract for cTokens 
+     * @param _cErc20Contract address of the Compound contract for cTokens
      * @param _numTokensToSupply number of erc20 tokens to deposit
      * @return uint256 0 on successful transaction, otherwise error code
-    */
+     */
     function depositErc20(
-        address _erc20Contract, 
-        address _cErc20Contract, 
+        address _erc20Contract,
+        address _cErc20Contract,
         uint256 _numTokensToSupply
     ) external returns (uint256) {
         // create references to the contracts on mainnet
-        Erc20 underlying = Erc20(_erc20Contract);
+        Erc20 token = Erc20(_erc20Contract);
         CErc20 cToken = CErc20(_cErc20Contract);
 
+        token.transferFrom(msg.sender, address(this), _numTokensToSupply);
+
         // Approve transfer on the ERC20 contract
-        underlying.approve(_cErc20Contract, _numTokensToSupply);
+        token.approve(_cErc20Contract, _numTokensToSupply);
```

Comment:
`Approve` first then `transferFrom`

Reply:
As in `approve` the contract to spend the user's tokens? Can we do that from inside this contract?

Reply:
It would be better to do this step using script.

Reply:
yes, we are approving the `_cErc20Contract ` to use funds